src/plugins/discover/public/application/components/context_app/use_context_app_fetch.tsx

```diff
@@ -0,0 +1,184 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useCallback, useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { fromPairs } from 'lodash';
+import { CONTEXT_TIE_BREAKER_FIELDS_SETTING } from '../../../../common';
+import { DiscoverServices } from '../../../build_services';
+import { fetchAnchorProvider } from '../../angular/context/api/anchor';
+import { EsHitRecord, fetchContextProvider, SurrDocType } from '../../angular/context/api/context';
+import { MarkdownSimple, toMountPoint } from '../../../../../kibana_react/public';
+import { IndexPattern, SortDirection } from '../../../../../data/public';
+import {
+  ContextFetchState,
+  FailureReason,
+  getInitialContextQueryState,
+  LoadingStatus,
+} from '../../angular/context_query_state';
+import { AppState } from '../../angular/context_state';
+import { getFirstSortableField } from '../../angular/context/api/utils/sorting';
+
+const createError = (statusKey: string, reason: FailureReason, error?: Error) => ({
+  [statusKey]: { value: LoadingStatus.FAILED, error, reason },
+});
+
+export interface ContextAppFetchProps {
+  anchorId: string;
+  indexPatternId: string;
+  indexPattern: IndexPattern;
+  appState: AppState;
+  useNewFieldsApi: boolean;
+  services: DiscoverServices;
+}
+
+export function useContextAppFetch({
+  anchorId,
+  indexPatternId,
+  indexPattern,
+  appState,
+  useNewFieldsApi,
+  services,
+}: ContextAppFetchProps) {
+  const { uiSettings: config, data, indexPatterns, toastNotifications, filterManager } = services;
+
+  const searchSource = useMemo(() => {
+    return data.search.searchSource.createEmpty();
+  }, [data.search.searchSource]);
+  const tieBreakerField = useMemo(
+    () => getFirstSortableField(indexPattern, config.get(CONTEXT_TIE_BREAKER_FIELDS_SETTING)),
+    [config, indexPattern]
+  );
+  const fetchAnchor = useMemo(() => {
+    return fetchAnchorProvider(indexPatterns, searchSource, useNewFieldsApi);
+  }, [indexPatterns, searchSource, useNewFieldsApi]);
+  const { fetchSurroundingDocs } = useMemo(
+    () => fetchContextProvider(indexPatterns, useNewFieldsApi),
+    [indexPatterns, useNewFieldsApi]
+  );
+
+  const [fetchedState, setFetchedState] = useState<ContextFetchState>(
+    getInitialContextQueryState()
+  );
+
+  const setState = useCallback((values: Partial<ContextFetchState>) => {
+    setFetchedState((prevState) => ({ ...prevState, ...values }));
+  }, []);
+
+  const fetchAnchorRow = useCallback(async () => {
+    const { sort } = appState;
+    const [[, sortDir]] = sort;
+    const anchorError = i18n.translate('discover.context.unableToLoadAnchorDocumentDescription', {
+      defaultMessage: 'Unable to load the anchor document',
+    });
+
+    if (!tieBreakerField) {
+      setState(createError('anchorStatus', FailureReason.INVALID_TIEBREAKER));
+      toastNotifications.addDanger({
+        title: anchorError,
```

Comment:
I left `tieBreakerField` chek for `fetchAnchorRow()` and removed from `fetchSurroundingRows()` due to:
- context rows requests (predecessors and successors) will be fired only after anchor document successfully retrived
- in case when we empty `context:tieBreakerFields` we need to refresh the page to apply changed settings, so anyway anchor request will be fired first